tf2onnx/onnx_opset/nn.py

```diff
@@ -1011,25 +1015,46 @@ def version_6(cls, ctx, node, **kwargs):
         is_training = node.get_attr_value('is_training', True)
 
         if is_training and node.get_attr_value('exponential_avg_factor', 1.0) == 1.0:
-            # Sometimes TF uses a BatchNorm op with training = True and exponential_avg_factor = 1.0
-            # to perform layer mean/variance normalization. In such cases, the mean/var are computed from the input.
-            # TF allows mean/variance to be excluded only if is_training and exponential_avg_factor == 1.0
-            utils.make_sure(inp_rank is not None, "Cannot convert node %s of type %s with input of unknown rank.",
-                            node.name, tf_type)
-            dims = [0] + list(range(2, inp_rank))
-            avg = GraphBuilder(ctx).make_reduce_mean({"data": node.input[0], "axes": dims, "keepdims": True})
-            avg_squeezed = GraphBuilder(ctx).make_squeeze({"data": avg, "axes": dims})
-            sub = ctx.make_node("Sub", [node.input[0], avg]).output[0]
-            var_squeezed = GraphBuilder(ctx).make_reduce_sum_square({"data": sub, "axes": dims, "keepdims": False})
-
-            inp_shape = ctx.make_node("Shape", [node.input[0]]).output[0]
-            dims_const = ctx.make_const(utils.make_name("axes_const"), np.array(dims, dtype=np.int64)).output[0]
-            reduce_dims = ctx.make_node("Gather", [inp_shape, dims_const]).output[0]
-            dims_product = GraphBuilder(ctx).make_reduce_prod({"data": reduce_dims, "axes": [0], "keepdims": False})
-            cnt_float = ctx.make_node("Cast", [dims_product], attr={'to': ctx.get_dtype(node.input[0])})
-
-            pop_var_squeezed = ctx.make_node("Div", [var_squeezed, cnt_float.output[0]]).output[0]
-            ctx.replace_inputs(node, node.input[:3] + [avg_squeezed, pop_var_squeezed])
+            if ctx.opset >= 17:
+                node.type = "LayerNormalization"
+                first_axis_to_norm = 2
```

Comment:
Hey @ShuaiShao93, why did you add this change? Don't we have to extract the axis value from the axis attribute already in the TF op?

Reply:
I don't quite understand this very well either. I just saw line 1035 seems to assume axis=2 is first_axis_to_norm. And I remember I could verify the parity end to end.

Reply:
BTW a better alternative is probably making https://github.com/onnx/tensorflow-onnx/blob/main/tf2onnx/rewriter/layer_normalization_rewriter.py work, so that we can also get rid of a bunch of other ops added by TF